### PR TITLE
Internal registry

### PR DESCRIPTION
**Pull request by tmbo**
- publish containers to internal docker registry
- publish to internal py registry

### CODE REVIEW

File: .github/workflows/continous-integration.yml [hunk 1]
```diff
@@ -1148,12 +1165,35 @@ jobs:
 
       - name: Build ⚒️ Distributions
         run: poetry build
+      
+      # TODO: readd release step to puplic pypi
```
Comment (m-vdb) [inline .github/workflows/continous-integration.yml:1169]: just a tiny suggestion to make this process easier :)
```suggestion
      # TODO: readd release step to puplic pypi
      # - name: Publish to PyPI 📦
      #    uses: pypa/gh-action-pypi-publish@717ba43cfbb0387f6ce311b169a825772f54d295
      #    with:
      #      user: __token__
      #      password: ${{ secrets.PYPI_TOKEN }}
```

**Review by m-vdb — APPROVED** [review]
thanks Tom 🙌🏻

**Review by miraai — APPROVED** [review]

**Review by miraai — APPROVED** [review]

**Comment by tmbo** [comment]
@miraai the docker pushing is finally working 🎉 but i have a permissions issue while pushing the pypi package to the internal google artifact registry:

```
ERROR: (gcloud.artifacts.print-settings.python) PERMISSION_DENIED: Permission 'artifactregistry.repositories.get' denied on resource '//artifactregistry.googleapis.com/projects/rasa-releases/locations/europe-west3/repositories/rasa-plus-py' (or it may not exist).
- '@type': type.googleapis.com/google.rpc.ErrorInfo
  domain: artifactregistry.googleapis.com
  metadata:
    permission: artifactregistry.repositories.get
    resource: projects/rasa-releases/locations/europe-west3/repositories/rasa-plus-py
  reason: IAM_PERMISSION_DENIED
```

The above error is from the tagged build `3.8.0a8.dev1`. the google auth uses `secrets.RASA_OSS_RELEASE_ACCOUNT_KEY` which is the same one that is used for [the niglty_release builds](https://github.com/RasaHQ/rasa/blob/2fdba86c1c7919b0206258352cc9e942a793f316/.github/workflows/nightly_release.yml#L101) so i hoped it'd just work. Do you have any idea what might be off with the permissions?

https://github.com/RasaHQ/rasa/actions/runs/6277978314/job/17051495441#step:12:193

**Review by miraai — APPROVED** [review]

**Comment by miraai** [comment]
> @miraai the docker pushing is finally working 🎉 but i have a permissions issue while pushing the pypi package to the internal google artifact registry:
> 
> ```
> ERROR: (gcloud.artifacts.print-settings.python) PERMISSION_DENIED: Permission 'artifactregistry.repositories.get' denied on resource '//artifactregistry.googleapis.com/projects/rasa-releases/locations/europe-west3/repositories/rasa-plus-py' (or it may not exist).
> - '@type': type.googleapis.com/google.rpc.ErrorInfo
>   domain: artifactregistry.googleapis.com
>   metadata:
>     permission: artifactregistry.repositories.get
>     resource: projects/rasa-releases/locations/europe-west3/repositories/rasa-plus-py
>   reason: IAM_PERMISSION_DENIED
> ```
> 
> The above error is from the tagged build `3.8.0a8.dev1`. the google auth uses `secrets.RASA_OSS_RELEASE_ACCOUNT_KEY` which is the same one that is used for [the niglty_release builds](https://github.com/RasaHQ/rasa/blob/2fdba86c1c7919b0206258352cc9e942a793f316/.github/workflows/nightly_release.yml#L101) so i hoped it'd just work. Do you have any idea what might be off with the permissions?
> 
> https://github.com/RasaHQ/rasa/actions/runs/6277978314/job/17051495441#step:12:193

Let me take a look

**Comment by miraai** [comment]
@tmbo I added write permissions for that service account, try again and let me know!

File: .github/workflows/continous-integration.yml [hunk 2]
```diff
@@ -1148,12 +1159,45 @@ jobs:
...
+        with:
+          credentials_json: "${{ secrets.RASA_OSS_RELEASE_ACCOUNT_KEY }}"
+
+      - name: "Set up Cloud SDK"
+        uses: "google-github-actions/setup-gcloud@62d4898025f6041e16b1068643bfc5a696863587"
 
-      - name: Publish to PyPI 📦
-        uses: pypa/gh-action-pypi-publish@c7f29f7adef1a245bd91520e94867e5c6eedddcc
+      - name: Setup tmate session
+        uses: mxschmitt/action-tmate@v3
```
Comment (rasa-jmac) [inline .github/workflows/continous-integration.yml:1190]: @tmbo could you pin this to a SHA please

Comment (tmbo) [reply .github/workflows/continous-integration.yml:1190]: this is just for debugging and will be removed before merge :)

**Comment by tmbo** [comment]
superseded by https://github.com/RasaHQ/rasa-private/pull/15

**Pull request closed without merging**